File: documentation-examples-scala/src/scala/dev/typr/foundations/docs/postgresql/StreamingInsertSingle.scala
package dev.typr.foundations.docs.postgresql

import dev.typr.scalafoundations.*

@SuppressWarnings(Array("unused"))
object StreamingInsertSingle:

  //start
  // Insert a list of strings using COPY
  def insertNames(names: Iterator[String], tx: Transactor): Long =
    streamingInsert
      .of("COPY users(name) FROM STDIN", 1000, names, PgTypes.text.pgText())
      .transact(tx)
  //stop

File: documentation-examples-java/src/java/dev/typr/foundations/docs/postgresql/StreamingInsertSingle.java
package dev.typr.foundations.docs.postgresql;

import dev.typr.foundations.*;
import java.util.Iterator;

@SuppressWarnings("unused")
public class StreamingInsertSingle {

  // start
  // Insert a list of strings using COPY
  long insertNames(Iterator<String> names, Transactor tx) {
    return StreamingInsert.of("COPY users(name) FROM STDIN", 1000, names, PgTypes.text.pgText())
        .transact(tx);
  }
  // stop
}
